cluster-autoscaler/utils/gpu/gpu.go

```diff
@@ -0,0 +1,42 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package gpu
+
+import (
+	apiv1 "k8s.io/api/core/v1"
+)
+
+const (
+	// ResourceNvidiaGPU is the name of the Nvidia GPU resource.
+	ResourceNvidiaGPU = "nvidia.com/gpu"
+)
+
+// SetGPUAllocatableToCapacity allows us to tolerate the fact that nodes with
+// GPUs can have allocatable set to 0 for multiple minutes after becoming ready
+// Without this workaround, Cluster Autoscaler will trigger an unnecessary
+// additional scale up before the node is fully uperational.
+// TODO: Remove this once we handle dynamically privisioned resources well.
+func SetGPUAllocatableToCapacity(nodes []*apiv1.Node) []*apiv1.Node {
```

Comment:
This function takes a list and returns a list of nodes, which to me suggests it shouldn't modify the original. However, this is a list of pointers and the function actually mutates the underlying objects, so:
1. There is no need to build a new list at all, since we're mutating the old one anyway.
1. I think it's confusing to do it that way.

Reply:
Does this look better?